apps/web/pages/api/book/recurring-event.ts

```diff
@@ -53,16 +56,34 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
       ...booking,
       appsStatus,
       allRecurringDates,
+      thirdPartyRecurringEventId,
+      numSlotsToCheckForAvailability,
       currentRecurringIndex: key,
       noEmail: key !== 0,
     };
 
-    const eachRecurringBooking = await handleNewBooking(recurringEventReq, {
-      isNotAnApiCall: true,
-      skipAvailabilityCheck: key >= numSlotsToCheckForAvailability,
-    });
+    const promiseEachRecurringBooking: ReturnType<typeof handleNewBooking> = handleNewBooking(
+      recurringEventReq,
+      {
+        isNotAnApiCall: true,
+        skipAvailabilityCheck: key >= numSlotsToCheckForAvailability || thirdPartyRecurringEventId !== null,
+      }
+    );
+
+    const eachRecurringBooking = await promiseEachRecurringBooking;
 
     createdBookings.push(eachRecurringBooking);
+
+    if (!thirdPartyRecurringEventId) {
```

Comment:
It would be `undefined` here as well for every iteration of non-recurring event supported calendars